src/circuits/sha256/sha256_ethereum.tcc

```diff
@@ -43,122 +48,577 @@ sha256_ethereum<FieldT>::sha256_ethereum(libsnark::protoboard<FieldT> &pb,
     // -- var tmp [64]byte
     // -- tmp[0] = 0x80
     // written in the checkSum function of the crypto/sha256 go package
-    libsnark::pb_variable_array<FieldT> length_padding =
-        from_bits({
-                // Total size of this vector = 512bits
-                // First part: 448bits <-> 56bytes
-                1,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-
-                // Last part: 64bits <-> 8bytes
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,0,0,
-                0,0,0,0,0,0,1,0,
-                0,0,0,0,0,0,0,0
-        },
-        ZERO
-    );
+    libsnark::pb_variable_array<FieldT> length_padding = from_bits(
+        {// Total size of this vector = 512bits
+         // First part: 448bits <-> 56bytes
+         1,
```

Comment:
I think this happens because of the value set to the `BinPackArguments` attribute in the config. I assume that to get the formatting we want, we need to declare the array first and pass it to the function after

Reply:
Thanks.  I missed this.

There seem to be a few factors.  If we remove those inline comments, it packs the values:
```
    libsnark::pb_variable_array<FieldT> length_padding = from_bits(                                                                                                   
        {1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0,                                                                                      
         0, 0, 0, 0, 0, 0, 0, 0},                                                                                                                                     
        ZERO);
```
And this seems to be the same if we move the array into its own variable.

To preserve the layout, I think we need to add the `//` at the end of every line (or use the `// clang-format off` pragma.

Reply:
With `//` at the end of every line, the format it retained.  If the inline comments are included on their own line, it puts one entry per-line again, so the best I can get seems to be:
```
    // Total size of this vector = 512bits                                                                                                                            
    const std::vector<bool> bit_list{                                                                                                                                 
        1, 0, 0, 0, 0, 0, 0, 0, // First part: 448bits <-> 56bytes                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        ....
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, // Last part: 64bits <-> 8bytes                                                                                                       
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 1, 0, //                                                                                                                                    
        0, 0, 0, 0, 0, 0, 0, 0, //                                                                                                                                    
    };                                                                                                                                                                
```

Reply:
I see, thanks